networkit/cpp/centrality/GroupClosenessLocalSearch.cpp

```diff
@@ -0,0 +1,1143 @@
+// networkit-format
+
+#include <algorithm>
+#include <atomic>
+#include <cassert>
+#include <cmath>
+#include <omp.h>
+#include <queue>
+#include <random>
+#include <stdexcept>
+#include <unordered_set>
+#include <utility>
+
+#ifdef __AVX2__
+#include <immintrin.h>
+#include <networkit/auxiliary/AlignedAllocator.hpp>
+#endif // __AVX2__
+
+#include <networkit/auxiliary/Parallel.hpp>
+#include <networkit/auxiliary/Random.hpp>
+#include <networkit/centrality/GroupClosenessGrowShrink.hpp>
+#include <networkit/centrality/GroupClosenessLocalSearch.hpp>
+#include <networkit/distance/Diameter.hpp>
+#include <networkit/graph/BFS.hpp>
+#include <networkit/graph/Dijkstra.hpp>
+
+#include <tlx/container/d_ary_addressable_int_heap.hpp>
+#include <tlx/container/d_ary_heap.hpp>
+#include <tlx/unused.hpp>
+
+namespace NetworKit {
+
+namespace {
+
+template <class WeightType>
+class GroupClosenessLocalSearchImpl final
+    : public GroupClosenessLocalSearch::GroupClosenessLocalSearchInterface {
+
+    static constexpr count K = 16; // Number of packed repetitions using AVX
+    static constexpr float maxInt16 = static_cast<float>(std::numeric_limits<uint16_t>::max());
+    static constexpr WeightType invalidDecr = static_cast<WeightType>(-1);
+    static constexpr WeightType infDistance = std::numeric_limits<WeightType>::max();
+
+public:
+    GroupClosenessLocalSearchImpl(const Graph &G, const std::vector<node> &initGroup,
+                                  bool runGrowShrink, count maxIterations);
+
+    void run() override;
+
+    ~GroupClosenessLocalSearchImpl() override = default;
+
+private:
+    const Graph *G;
+    const bool runGrowShrink;
+    const count maxIterations;
+
+    std::vector<WeightType> distance, distance2;
+    std::vector<WeightType> oldDistance, oldDistance2;
+    std::vector<std::vector<WeightType>> distanceGlobal, distance2Global;
```

Comment:
For several of these vectors, it's not clear what we store here (especially the `std::vector<std::vector<...>>` variables). Add comments.